ClassNotFoundException on primitive arguments with quarkus-test-security

**Describe the bug**
If I include quarkus-test-security, a ClassNotFoundException is raised if there are test methods with arguments of primitive data types (for instance on a `@ParametrizedTest`). No need to actually use the `@TestSecurity` annotation.

**Expected behavior**
No exception is raised.

**Actual behavior**
```
java.lang.reflect.InvocationTargetException
	...
	at io.quarkus.test.junit.QuarkusTestExtension.beforeEach(QuarkusTestExtension.java:363)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachCallbacks$1(TestMethodTestDescriptor.java:161)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$5(TestMethodTestDescriptor.java:197)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	...
Caused by: java.lang.RuntimeException: Unable to setup @TestSecurity
	at io.quarkus.test.security.QuarkusSecurityTestExtension.beforeEach(QuarkusSecurityTestExtension.java:60)
	... 103 more
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: int
	at io.quarkus.test.security.QuarkusSecurityTestExtension.lambda$beforeEach$0(QuarkusSecurityTestExtension.java:34)
        ...
	at io.quarkus.test.security.QuarkusSecurityTestExtension.beforeEach(QuarkusSecurityTestExtension.java:36)
	... 103 more
Caused by: java.lang.ClassNotFoundException: int
        ...
	at io.quarkus.test.security.QuarkusSecurityTestExtension.lambda$beforeEach$0(QuarkusSecurityTestExtension.java:32)
	... 111 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. Add quarkus-test-security dependency
    ```
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-test-security</artifactId>
      <scope>test</scope>
    </dependency>
    ```
2. Create a ParametrizedTest with arguments of primitive data types:
    ```
    @QuarkusTest
    class QuarkusSecurityBugTest {
    
        @ParameterizedTest
        @ValueSource(ints = 1)
        public void replayTest(int number) {
        }
    }
    ```

**Environment (please complete the following information):**
 - Quarkus version: 1.8.1.Final

--- a/integration-tests/elytron-resteasy/src/test/java/io/quarkus/it/resteasy/elytron/TestSecurityTestCase.java
+++ b/integration-tests/elytron-resteasy/src/test/java/io/quarkus/it/resteasy/elytron/TestSecurityTestCase.java
@@ -2,9 +2,14 @@ package io.quarkus.it.resteasy.elytron;
 
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.stream.Stream;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
 
 import io.quarkus.test.junit.QuarkusTest;
@@ -28,6 +33,17 @@ class TestSecurityTestCase {
     @ParameterizedTest
     @ValueSource(ints = 1) //https://github.com/quarkusio/quarkus/issues/12413
     void testGetWithSecEnabled(int ignore) {
+        given()
+                .when()
+                .get("/secure")
+                .then()
+                .statusCode(401);
+    }
+
+    @TestSecurity
+    @ParameterizedTest
+    @MethodSource("arrayParams") //https://github.com/quarkusio/quarkus/issues/12413
+    void testGetUnAuthorized(int[] ignoredPrimitives, String[] ignored) {
         given()
                 .when()
                 .get("/secure")
@@ -64,6 +80,11 @@ class TestSecurityTestCase {
                 .then()
                 .statusCode(200)
                 .body(is("testUser"));
+    }
+
+    static Stream<Arguments> arrayParams() {
+        return Stream.of(
+                arguments(new int[] { 1, 2 }, new String[] { "hello", "world" }));
     }
 
 }

--- a/integration-tests/elytron-resteasy/src/test/java/io/quarkus/it/resteasy/elytron/TestSecurityTestCase.java
+++ b/integration-tests/elytron-resteasy/src/test/java/io/quarkus/it/resteasy/elytron/TestSecurityTestCase.java
@@ -4,6 +4,8 @@ import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.is;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.security.TestSecurity;
@@ -22,9 +24,10 @@ class TestSecurityTestCase {
                 .body(is("secure"));
     }
 
-    @Test
     @TestSecurity
-    void testGetWithSecEnabled() {
+    @ParameterizedTest
+    @ValueSource(ints = 1) //https://github.com/quarkusio/quarkus/issues/12413
+    void testGetWithSecEnabled(int ignore) {
         given()
                 .when()
                 .get("/secure")

--- a/test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java
+++ b/test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java
@@ -28,6 +28,9 @@ public class QuarkusSecurityTestExtension implements QuarkusTestBeforeEachCallba
             Class<?> original = cl.loadClass(context.getTestMethod().getDeclaringClass().getName());
             Method method = original.getDeclaredMethod(context.getTestMethod().getName(),
                     Arrays.stream(context.getTestMethod().getParameterTypes()).map(s -> {
+                        if (s.isPrimitive()) {
+                            return s;
+                        }
                         try {
                             return cl.loadClass(s.getName());
                         } catch (ClassNotFoundException e) {

--- a/test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java
+++ b/test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java
@@ -32,7 +32,7 @@ public class QuarkusSecurityTestExtension implements QuarkusTestBeforeEachCallba
                             return s;
                         }
                         try {
-                            return cl.loadClass(s.getName());
+                            return Class.forName(s.getName(), false, cl);
                         } catch (ClassNotFoundException e) {
                             throw new RuntimeException(e);
                         }

